spreg/sur_utils.py

```diff
@@ -39,7 +38,10 @@ def sur_dictxy(db,y_vars,x_vars,space_id=None,time_id=None):
     c = "Constant"
     if (len(y_vars) > 1):    # old format
         n_eq = len(y_vars)
-        y = np.array([db.by_col(name) for name in y_vars]).T
+        try:
+            y = np.array([db[name] for name in y_vars]).T
```

Comment:
These constructions could be a little simpler:

```python
if isinstance(db, libpysal.io.FileIO):
    y = db.by_col_array(*y_vars)
else:
    y = db[y_vars].values
```

But I'm ok with this try/except pattern if that's easier to understand for others. Certainly not worth delaying the release for this.